ls1intum/Artemis
# Allow multiple submissions in text exercises

### Checklist

### Motivation and Context


We want to allow students to submit text exercises multiple times.
The criteria for allowing to submit an exercise are the following:
* There is no due date
* The due date is before the current date
* The participation initialization date is after the due date

### Description

The submit button and text area will get locked as soon as the due date passed.The server will also return a forbidden in these cases. I also added new tests for this behavior.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new text exercise with a due date
4. You should be able to participate and submit multiple times
5. As soon as the due date passed, the submit button and the text area should get locked and the tooltips should get updated

### Screenshots
![text_multiple_submit](https://user-images.githubusercontent.com/27979674/68246995-3796c980-001a-11ea-841e-0ed773896d62.gif)
